google-cloud-spanner/lib/google/cloud/spanner/commit.rb

```diff
@@ -327,14 +327,14 @@ def key_set keys
           keys = [keys] unless keys.is_a? Array
           return Google::Spanner::V1::KeySet.new(all: true) if keys.empty?
           if keys_are_ranges? keys
-            keys = [keys] unless keys.is_a? Array
             key_ranges = keys.map do |r|
               Convert.to_key_range(r)
             end
             return Google::Spanner::V1::KeySet.new(ranges: key_ranges)
           end
-          key_list = Array(keys).map do |i|
-            Convert.raw_to_value(Array(i)).list_value
+          key_list = keys.map do |key|
+            key = [key] unless key.is_a? Array
```

Comment:
I don't get it, why is each key being wrapped in an array?

Reply:
The key, whether a table primary key, or an index key, can be one or many values. So the format for specifying an individual key is a list. This is how Spanner represents this.

Reply:
Great, thanks.